apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+import React, {PropTypes} from 'react';
+import $ from 'jquery';
+import i18n from '@cdo/locale';
+import color from '../../util/color';
+import BaseDialog from '../../templates/BaseDialog';
+import Button from '../../templates/Button';
+import SchoolInfoInputs, {SCHOOL_TYPES_HAVING_NCES_SEARCH} from '../../templates/SchoolInfoInputs';
+
+const styles = {
+  container: {
+    margin: 20,
+    color: color.charcoal
+  },
+  heading: {
+    fontSize: 16,
+    fontFamily: "'Gotham 5r', sans-serif",
+  },
+  middle: {
+    marginTop: 20,
+    marginBottom: 20,
+    paddingBottom: 20,
+    paddingTop: 20,
+    borderTopWidth: 1,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.lighter_gray,
+  },
+  bottom: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  error: {
+    color: color.red,
+  },
+};
+
+export default class SchoolInfoInterstitial extends React.Component {
+  static propTypes = {
+    // This component is tightly bound to the HAML view that renders it and
+    // populates its props, and similarly to the User update API that
+    // it uses to save entered school info.
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        // Note, these names intentionally match the fields on the SchoolInfo
+        // model, not their JavaScript-friendly equivalents.  The mapping
+        // occurs in the constructor, below, and back when we submit.
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        state: PropTypes.string,
+        zip: PropTypes.string,
+        full_address: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onClose: function () {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {existingSchoolInfo} = this.props.scriptData;
+    const initialNcesSchoolId = existingSchoolInfo.school_id ?
+      existingSchoolInfo.school_id :
+      (
+        existingSchoolInfo.country === 'United States'
```

Comment:
I'm pretty sure that you also need to check `existingSchoolInfo.country === 'United States'`
We seem to be inconsistent about how we specify the country in `school_info`. Sometimes it is set to a two-character country code and sometimes to the country name. The validation logic doesn't check this and the column is longer than two characters.

Reply:
OK, it looks like you actually map `'US'` to `'United States'` in `_school_info_interstitial.html.haml`.
Perhaps add a comment here to call that out. Also, if we are doing that mapping in the haml then why not also do the mapping of `school_id` to `"-1"` there?

Reply:
That's a good point.

I added this code because I discovered a funny edge case for partial entry: We never send/save a school id of `"-1"` to the database, so there's nothing to differentiate a form submitted where there's been no interaction with the school autocomplete dropdown (`school_id` is `null`) and one where the user has checked "I can't find my school" but hasn't entered any details yet (`school_id` is `"-1"`) - in essence, we don't serialize the state of that checkbox.

As much as possible, I'd like all the business logic around this component to live in one layer, and the checkbox state in particular seems like a UI concern so I think it should live with the React component. So I'm leaning toward moving the `US/United States` logic into the React component as well.  Does that sound okay to you?

Reply:
Probably easier to test that stuff in the React layer than in a HAML file, too.